### PR TITLE
Bump k8s version to v1.24.0-alpha.2

### PR DESCRIPTION
**Pull request by nckturner**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Bump kubernetes dependencies to v1.24.0-alpha.2

**Special notes for your reviewer**:
/assign @wongma7 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/e2e/run.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ UP="${UP:-yes}"
 # if DOWN==yes, delete cluster after test
 DOWN="${DOWN:-yes}"
 
-KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.23.2}"
+KUBERNETES_VERSION="${KUBERNETES_VERSION:-v0.24.0-alpha.3}"
```
Comment (wongma7) [inline hack/e2e/run.sh:50]: should it be 1.24?

Comment (wongma7) [reply hack/e2e/run.sh:50]:
```

I0318 22:24:39.156681   25487 populate_instancegroup_spec.go:308] Cannot set default Image for CloudProvider="aws"
Error: unable to determine default image for InstanceGroup master-us-west-2a
Error: exit status 1 
```
does kops support creating such clusters yet?

Comment (nckturner) [reply hack/e2e/run.sh:50]: Yeah I was wondering if that would work :)

**Comment by wongma7** [comment]
/lgtm
/hold

need to create release-1.23 branch, IMO should do it before merge otherwise we will forget : ) (see: missing release-1.21 branch)

**Comment by nckturner** [comment]
```
which golint 2>&1 >/dev/null || go get golang.org/x/lint/golint
go: downloading golang.org/x/lint v0.0.0-20210508222113-6edffad5e616
go: downloading golang.org/x/tools v0.1.8
golint -set_exit_status k8s.io/cloud-provider-aws/cmd/aws-cloud-controller-manager k8s.io/cloud-provider-aws/cmd/ecr-credential-provider k8s.io/cloud-provider-aws/pkg/apis/config k8s.io/cloud-provider-aws/pkg/apis/config/v1alpha1 k8s.io/cloud-provider-aws/pkg/providers/v1 k8s.io/cloud-provider-aws/pkg/providers/v2 k8s.io/cloud-provider-aws/pkg/providers/v2/mocks
/bin/bash: golint: command not found
make: *** [Makefile:94: verify-lint] Error 127
+ EXIT_VALUE=2
+ set +o xtrace
```

**Comment by nckturner** [comment]
Ah its probably a `go get` vs `go install` sort of thing...

**Pull request merged**